modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -492,17 +489,52 @@ void cv::gimpl::ie::GIEExecutable::run(std::vector<InObj>  &&input_objs,
 namespace cv {
 namespace gimpl {
 namespace ie {
+inline int getIdxByName(const std::vector<std::string>& reshape_names, const std::string& name) {
+    const auto it = std::find(reshape_names.begin(), reshape_names.end(), name);
+    GAPI_Assert(it != reshape_names.end());
+    return std::distance(reshape_names.begin(), it);
+}
 
-static void configureInputInfo(const IE::InputInfo::Ptr& ii, const cv::GMetaArg mm) {
+inline void configureReshape(const IE::InputInfo::Ptr& ii, IEUnit* const uu, const std::vector<int>& image_size) {
+    const auto layout = ii->getLayout();
+    size_t to_hw_shift = 0;
+    switch (layout) {
+        case InferenceEngine::NHWC:
+        {
+            to_hw_shift = 1;
+            break;
+        }
+        case InferenceEngine::NCHW:
+        {
+            break;
+        }
+        default:
+            util::throw_error(std::runtime_error("Unsupported input layout for IE backend's reshape"));
+    }
+    auto input_dims = ii->getTensorDesc().getDims();
+    std::copy(image_size.begin(), image_size.end(), input_dims.end() - image_size.size() - to_hw_shift);
+    const auto input_name = ii->name();
+    const auto idx = getIdxByName(uu->params.reshape_in_names, input_name);
+    uu->params.reshape_in_shapes[uu->params.reshape_in_names.at(idx)] = input_dims;
+}
```

Comment:
Can you please explain what happens here in this function?

Reply:
Added comments about steps.